### ISSUE TITLE
Document all Astronomer Helm configs

### ISSUE THREAD

**Issue by jwitz** [astronomer/docs#239]
@jwitz commented on [Mon Dec 21 2020](https://github.com/astronomer/docs-archived/issues/161)

@sjmiller609 commented on [Tue Jul 28 2020](https://github.com/astronomer/issues/issues/1601)

**Describe the document**

We should have a reference document for all Astronomer configuration options. A configuration option should either be unsupported or documented. This includes all helm, houston, airflow helm, and airflow configuration options.

**Information**

Check all that apply:
- [x] Was this directly requested by a customer? If so, which one?
- [x] Does this relate to a Cloud use case?
- [x] Does this relate to an Enterprise use case? 

Our support team has a lot of difficulty finding if options exist or not in the Astronomer platform.

**User Story**

Reference material for installation or reconfiguration.
"As an astronomer user, how can I configure the platform"
"As an astronomer support engineer, how can I find a list of the available options"

**Screenshots**

**Alternatives**

Users can manually comb the helm values in the umbrella chart, each subchart, the airflow chart, and the houston config.

---

@sjmiller609 commented on [Tue Jul 28 2020](https://github.com/astronomer/issues/issues/1601#issuecomment-665157857)

I think this one is hard because it involves knowing what every feature does. I think the infra and devops team should handle it so we can dig down and find all configurations and add a check in CI to ensure docs are added when new configs are created.

---

@schnie commented on [Tue Jul 28 2020](https://github.com/astronomer/issues/issues/1601#issuecomment-665219608)

I think a baseline thing we can do is what we also _started_ over on the airflow chart - https://github.com/astronomer/airflow-chart#parameters. This is a pretty typical thing helm chart doc format. Could have higher-level content on our docs that connect the dots and create recipes, etc.

---

@jpweber commented on [Tue Aug 04 2020](https://github.com/astronomer/issues/issues/1601#issuecomment-668831790)

https://github.com/norwoodj/helm-docs this will handle the initial generation of the helm values for us. 

Which raises a question, do we want all the Chart/Subchart values on one page or should the values of a sub chart live in the readme of the sub chart? 

I personally kind of prefer having them all together but I can see a good case for both ways.

**Comment by kwcanuck** [comment]
@jwitz this is another oldie! : )

It would appear that you and Paola made significant progress in the past two years, and I'm not sure if this issue is still relevant. 

Temporarily assigning the issue to you for review and analysis. Thanks!

**Comment by jwitz** [comment]
There's been a lot of debate about how much of the Helm chart to configure. We've made less progress towards a comprehensive Helm reference guide, which seems unproductive given that a lot of values are commented in the physical Helm chart. Rather, we've focused on making sure that important Helm configurations are documented and provided context for real world setups, like in https://docs.astronomer.io/software/export-task-logs. Because there's no urgency to fully document the Helm chart, we can close this issue.
